src/libp2p/metrics.rs

```diff
@@ -33,3 +37,33 @@ pub static BAD_PEERS: Lazy<Gauge> = Lazy::new(|| {
     );
     metric
 });
+
+pub static PEER_TIPSET_EPOCH: Lazy<Family<PeerLabel, Gauge>> = Lazy::new(|| {
+    let metric = Family::default();
+    crate::metrics::default_registry().register(
+        "peer_tipset_epoch",
+        "peer tipset epoch",
+        metric.clone(),
+    );
+    metric
+});
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct PeerLabel(PeerId);
+
+impl PeerLabel {
+    pub const fn new(peer: PeerId) -> Self {
+        Self(peer)
+    }
+}
+
+impl EncodeLabelSet for PeerLabel {
+    fn encode(&self, mut encoder: LabelSetEncoder) -> Result<(), std::fmt::Error> {
+        let mut label_encoder = encoder.encode_label();
+        let mut label_key_encoder = label_encoder.encode_label_key()?;
+        EncodeLabelKey::encode(&"PEER", &mut label_key_encoder)?;
```

Comment:
can this not be:

``` "PEER".encode(&mut label_key_encoder)```?

Reply:
It can't, but it seems we can do `EncodeLabelKey::encode("PEER", &mut label_key_encoder)?;`

Reply:
It does not work.
```
EncodeLabelKey::encode("libp2p_message_kind", &mut label_key_encoder)?;
    |         ---------------------- ^^^^^^^^^^^^^^^^^^^^^ the trait `prometheus_client::encoding::EncodeLabelKey` is not implemented for `str`
    |         |
    |         required by a bound introduced by this call
```

Reply:
A side note, I don't think ` "PEER".encode(&mut label_key_encoder)` would improve readability when the trait is not well-known. The idea is similar to why we favor `Arc::clone(t)` over `t.clone` in the code base, see https://github.com/ChainSafe/forest/pull/4039#issuecomment-1983686248

> Using Arc::clone makes it explicit that you want to clone the Arc and not the underlying object. It makes the code easier to understand and harder to make mistakes.